code/obj/item/gun/energy.dm

```diff
@@ -1839,3 +1839,270 @@ TYPEINFO(/obj/item/gun/energy/wasp)
+			heat += FIRE_THRESHOLD // spicy!
+			update_icon()
+
+
+	emp_act()
+		if (our_cell)
+			src.visible_message(SPAN_ALERT("[src]'s cell violently overheats!"))
+			src.add_heat(FIRE_THRESHOLD)
+
+	New()
+		processing_items |= src
+		set_current_projectile(new/datum/projectile/laser/makeshift)
+		projectiles = list(current_projectile)
+		..()
+
+	Exited(Obj, newloc)
+		var/obj/item/cell/C = Obj
+		if (istype(C) && !QDELETED(C))
+			C.update_icon()
+			var/datum/component/power_cell/comp = C.GetComponent(/datum/component/power_cell)
+			comp.UnregisterFromParent()
+			comp.RemoveComponent()
+			our_cell = null
+			update_icon()
+		. = ..()
+
+
+	process()
+		if (heat > 0)
+			if (heat > FIRE_THRESHOLD)
+				var/mob/living/victim = src.loc
+				if (istype(victim))
+					victim.changeStatus("burning", 7 SECONDS)
+					if (!ON_COOLDOWN(victim, "makeshift_burn", 5 SECONDS))
+						boutput(victim, SPAN_ALERT("You are set on fire due to the extreme temperature of [src]!"))
+						victim.emote("scream")
+			heat = max(0, heat - HEAT_REMOVED_PER_PROCESS)
+			update_icon()
+		return
+
+	canshoot(mob/user)
+		if (heat_repair != 0)
+			boutput(user,SPAN_ALERT("[src] will need repairs before being able to function!"))
+			return FALSE
+		if (!our_light)
+			boutput(user,SPAN_ALERT("[src] needs a light source to function!"))
+			return FALSE
+		else if (our_light.light_status != LIGHT_OK)
+			boutput(user,SPAN_ALERT("[src] has no reaction when you pull the trigger!"))
+			return FALSE
+		else
+			return ..()
+
+	attackby(obj/item/W, mob/user, params)
+		if (heat < FIRE_THRESHOLD)
+			if(heat_repair) // gun machine broke, we need to repair it
+				if (issnippingtool(W) && heat_repair == 1)
+					boutput(user,SPAN_NOTICE("You remove the burnt wiring from [src]."))
+					playsound(src, 'sound/items/Wirecutter.ogg', 50, TRUE)
+					heat_repair++
+					src.icon_state = "makeshift-burnt-2"
+					update_icon()
+					return
+				else if (istype(W, /obj/item/cable_coil) && heat_repair == 2)
+					if (W.amount >= 10)
+						SETUP_GENERIC_ACTIONBAR(user, src, 3 SECONDS, /obj/item/gun/energy/makeshift/proc/finish_repairs,\
+						list(W,user), W.icon, W.icon_state, SPAN_NOTICE("[user] replaces the burnt wiring within [src]."), null)
+					else
+						boutput(user,SPAN_NOTICE("You need at least 10 wire to repair the wiring."))
+					return
+			else if (iswrenchingtool(W) && our_cell)
+				var/obj/item/removed_cell = our_cell
+				SEND_SIGNAL(src, COMSIG_CELL_SWAP, null)
+				boutput(user,SPAN_NOTICE("You disconnect [our_cell] from [src]."))
+				playsound(src, 'sound/items/Ratchet.ogg', 50, TRUE)
+				user.put_in_hand_or_drop(removed_cell)
+				return
+			else if (istype(W, /obj/item/cell) && !our_cell)
+				user.u_equip(W)
+				boutput(user,SPAN_NOTICE("You attach [W] to [src]."))
+				attach_cell(W, user)
+				return
+			else if (issnippingtool(W) && our_light)
+				boutput(user,SPAN_NOTICE("You remove [our_light] from the barrel."))
```

Comment:
I'd prefer this refer to cutting away the wiring attaching the light.  Otherwise I don't understand how you remove it with a snipping implement.